core/platform/bulk_email/mailchimp_bulk_email_services.py

```diff
@@ -0,0 +1,171 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for handling mailchimp API calls."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import ast
+import hashlib
+
+import feconf
+import mailchimp3
+from mailchimp3 import mailchimpclient
+
+import python_utils
+
+
+def _get_subscriber_hash(email):
+    """Returns Mailchimp subscriber hash from email.
+
+    Args:
+        email: str. The email of the user.
+
+    Raises:
+        Exception. Invalid type for email, expected string.
+
+    Returns:
+        str. The subscriber hash corresponding to the input email.
+    """
+    if not isinstance(email, python_utils.BASESTRING):
+        raise Exception(
+            'Invalid type for email. Expected string, received %s' % email)
+    md5_hash = hashlib.md5()
+    md5_hash.update(email)
+    return md5_hash.hexdigest()
+
+
+def _get_mailchimp_class():
+    """Returns the mailchimp api class. This is separated into a separate
+    function to facilitate testing.
+
+    NOTE: No other functionalities should be added to this function.
+
+    Raises:
+        Exception. Mailchimp API key is not available.
+        Exception. Mailchimp username is not set.
+
+    Returns:
+        Mailchimp. A mailchimp class instance with the API key and username
+        initialized.
+    """
+    if not feconf.MAILCHIMP_API_KEY:
+        raise Exception('Mailchimp API key is not available.')
+
+    if not feconf.MAILCHIMP_USERNAME:
+        raise Exception('Mailchimp username is not set.')
+
+    return mailchimp3.MailChimp(
+        mc_api=feconf.MAILCHIMP_API_KEY, mc_user=feconf.MAILCHIMP_USERNAME)
+
+
+def permanently_delete_user_from_list(user_email):
+    """Permanently deletes the user with the given email from the Mailchimp
+    list.
+
+    NOTE: This should only be called from the wipeout service since once a user
```

Comment:
Perhaps `NOTE:` --> `NOTE TO DEVELOPERS:` (so that it stands out a bit more)

Reply:
Done